architecture/environments/prod/main.tf

```diff
@@ -52,6 +52,15 @@ module "user_upload_files_bucket" {
   depends_on       = [module.requirements]
 }
 
+module "user_datasets_bucket" {
```

Comment:
Any benefit to separate buckets vs. sharing one with specific prefixes?

Reply:
In my mind it allows potential different security rules, a cleaner hierachy and less chance of getting something wrong/overwriting a blob's path. Also afaik there's no extra cost.